Lib/http/client.py

```diff
@@ -181,11 +181,10 @@ def _strip_ipv6_iface(enc_name: bytes) -> bytes:
     return enc_name
 
 class HTTPMessage(email.message.Message):
-    # XXX The only usage of this method is in
-    # http.server.CGIHTTPRequestHandler.  Maybe move the code there so
-    # that it doesn't need to be part of the public API.  The API has
-    # never been defined so this could cause backwards compatibility
-    # issues.
+
+    # The getallmatchingheaders() method was only used by the CGI handler
+    # that was removed in Python 3.15. However, since the public API was not
+    # properly defined, it will be kept for backwards compatibility reasons.
```

Comment:
Hmm, perhaps a candidate for deprecation in 3.15? Separate issue to check for usage and so on. If so, perhaps can be removed in 3.17 rather than the 3.20.

Reply:
We have a separate issue for this method so we can discuss its removal as well. I'll need to find it though.